File: implementations/kotlin-springboot/harness/test/testdata/no-cross-bc-domain-import/bad-card-imports-account-domain/card/domain/Card.kt
package com.example.accountservice.card.domain

// Violation — directly imports another BC's(account) domain/ package. Only ID references like
// accountId: String are allowed(tactical-ddd.md).
import com.example.accountservice.account.domain.Account

class Card private constructor() {
    var cardId: String = ""
        private set

    var account: Account? = null
        private set

    companion object {
        fun create(account: Account): Card =
            Card().apply {
                this.cardId = "card-1"
                this.account = account
            }
    }
}

File: implementations/fastapi/harness/tests/fixtures/no-cross-bc-domain-import/bad-imports-other-bc-domain/src/card/domain/card.py
from __future__ import annotations

from datetime import datetime

from ...payment.domain.payment import Payment


class Card:
    def __init__(self, card_id: str, account_id: str, created_at: datetime, payment: Payment | None = None) -> None:
        self.card_id = card_id
        self.account_id = account_id
        self.created_at = created_at
        self.payment = payment
